### DISCUSSION TITLE
EMQX TLS/SSL failure on Version upgrade

### DISCUSSION THREAD

**Discussion by akshayparseja** [emqx/emqx#14363] [Q&A] [answered]
Hi team,
I am upgrading my emqx from 5.0.23 to 5.7.2.
In this upgrade i have everything working which includes acl , http auth but all my clients which connect perfectly fine in my old emqx endup in ssl error..

 {esockd_connection_sup,<0.6945.0>}, errorContext: ssl_error, reason: {tls_alert,{unexpected_message,"TLS server: In state wait_cert at ssl_gen_statem.erl:803 generated SERVER ALERT: Fatal - Unexpected Message\n {unexpected_msg,{internal,{change_cipher_spec,<<1>>}}}"}}, offender: [{pid,<0.6945.0>},{name,connection},{mfargs,{emqx_connection,start_link,[#{listener => {ssl,default},limiter => #{connection => #{initial => 0,rate => infinity,burst => 0}},zone => external,enable_authn => true}]}}]
 
 
 on reverting back to 5.0.23 without any config changes to ssl listener or the certs, they all connect and my system works just fine.
 
Can you help understanding why so.

On disabling verify peer, all my conns are back so the only blocker for my version upgrade is the ssl config

**Comment by zmstone** [comment]
Hi @akshayparseja 
Although it's the server emitting the fatal alert, but which client SDK is it ?

**Comment by zmstone** [comment, marked as answer]
Found similar error reported to Erlang/OTP.
I will link the issue to this discussion once confirmed:
1. Your MQTT client is running on Erlang/OTP, version 24
2. The issue can be fixed by configuring tls versions to allow only `tls1.2`, (do not add `tls1.3` to the list)

**Reply by akshayparseja** [reply]
Adding 1.2 only worked , thanks!
